content/app/app-monitoring.md

```diff
@@ -171,12 +181,135 @@ When creating GrafanaDashboard, you have to specify Grafana instance selector, s
 dashboard in Grafana instance:
 
 ```yaml
+apiVersion: grafana.integreatly.org/v1beta1
+kind: GrafanaDashboard
+metadata:
+  name: my-grafana-dashboard
+spec:
   instanceSelector:
     matchLabels:
       grafana: grafana-{{ your-tenant-name }}
+  .
+  .
+  .
 ```
 
 In addition, in case your GrafanaDashboard is not in the same namespace as your monitoring stack, you have to specify an additional field:
 ```yaml
+apiVersion: grafana.integreatly.org/v1beta1
+kind: GrafanaDashboard
+metadata:
+  name: my-grafana-dashboard
+spec:
+  instanceSelector:
+    matchLabels:
+      grafana: grafana-{{ your-tenant-name }}
+  allowCrossNamespaceImport: "true"
+  .
+  .
+  .
+```
+
+
+## Creating Grafana Datasources
+
+To create Grafana Datasources you have to create `GrafanaDatasource` CR.
+If you create it directly with UI, the changes will not be persisted for long,
+so it's advised to use Grafana UI for designing your Dashboard and then export it to CR.
+
+When creating GrafanaDatasource, you have to specify Grafana instance selector, so Grafana Operator can inject your
+dashboard in Grafana instance:
+
+```yaml
+apiVersion: grafana.integreatly.org/v1beta1
+kind: GrafanaDatasource
+metadata:
+  name: my-datasource
+spec:
+  instanceSelector:
+    matchLabels:
+      grafana: grafana-{{ your-tenant-name }}
+  .
+  .
+  .
+```
+
+In addition, in case your GrafanaDatasource is not in the same namespace as your monitoring stack, you have to specify an additional field:
+```yaml
+apiVersion: grafana.integreatly.org/v1beta1
+kind: GrafanaDatasource
+metadata:
+  name: my-datasource
+spec:
+  instanceSelector:
+    matchLabels:
+      grafana: grafana-{{ your-tenant-name }}
   allowCrossNamespaceImport: "true"
+  .
+  .
+  .
+```
+
+
+## Creating Grafana Plugins for Dashboards/Datasources
```

Comment:
"Creating" -> "Using"

When I see "Creating" I expect to see plugin implementation or something like this, which is not what the section about